catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -211,34 +154,87 @@ public void setEnableBackupPlugin(boolean enablePlugin) {
+        LOGGER.trace("Set root backup directory to: {}", this.rootBackupDir.toString());
+    }
+
+    ExecutorService getExecutor() {
+        return executor;
+    }
+
+    void setExecutor(ExecutorService executor) {
+        this.executor = executor;
+    }
+
+    private synchronized void execute(Runnable task) {
+
+        executor.execute(task);
+    }
+
+    private void create(List<Metacard> metacards) {
+
+        List<String> errors = new ArrayList<>();
+        for (Metacard metacard : metacards) {
+            try {
+                createFile(metacard);
+            } catch (RuntimeException | IOException e) {
+                errors.add(metacard.getId());
+            }
         }
 
-        this.rootBackupDir = new File(dir);
-        LOGGER.debug("Set root backup directory to: {}", this.rootBackupDir.toString());
+        if (!errors.isEmpty()) {
+            LOGGER.debug(getExceptionMessage(errors, CREATE));
+        }
     }
 
-    /**
-     * Sets the number of subdirectory levels to create. Two characters from
-     * each metacard ID will be used to name each subdirectory level.
-     *
-     * @param levels number of subdirectory levels to create
-     */
-    public void setSubDirLevels(int levels) {
-        this.subDirLevels = levels;
-        LOGGER.debug("Set subdirectory levels to: {}", this.subDirLevels);
+    private void delete(List<Metacard> cards) {
+        List<String> errors = new ArrayList<>();
+        for (Metacard metacard : cards) {
+            try {
+                deleteFile(metacard);
+            } catch (IOException e) {
+                errors.add(metacard.getId());
+            }
+        }
+
+        if (errors.size() > 0) {
+            LOGGER.debug(getExceptionMessage(errors, DELETE));
+        }
     }
 
-    private void backupMetacard(Metacard metacard) throws IOException {
+    private void removeTempExtension(File source) throws IOException {
+        LOGGER.trace("Removing {} file extension.", TEMP_FILE_EXTENSION);
+        File destination = new File(StringUtils.removeEnd(source.getAbsolutePath(),
+                TEMP_FILE_EXTENSION));
+        boolean success = source.renameTo(destination);
+        if (success) {
+            LOGGER.trace("Moved {} to {}.",
+                    source.getAbsolutePath(),
+                    destination.getAbsolutePath());
+        } else {
+            LOGGER.trace("Failed to move {} to {}.",
+                    source.getAbsolutePath(),
+                    destination.getAbsolutePath());
+        }
+    }
+
+    private String getExceptionMessage(List<String> metacardsIdsInError, String operation) {
+
+        return "Catalog Backup Plugin processing error." + " " + "Unable to " + operation
+                + " metacard(s) [" + StringUtils.join(metacardsIdsInError, ",") + "]. ";
+    }
+
+    private void createFile(Metacard metacard) throws IOException {
 
         // Write metacard to a temp file. When write is complete, rename (remove
```

Comment:
I have no idea, which is why I hoped you could provide an explanation :-) The only thing I can think of is to prevent two concurrent operations from writing to the same file at the same time. But then again, since the metacard ID is used to generate the file name, they would still both write to the same file. So no clue :-/